MetaMask/metamask-mobile
# feat: nft + swap metrics

## **Description**

Adds metrics for:
- Removing an NFT
- Opening details for an NFT

Improves the `source` parameter for the `SWAPS_OPENED` metric.  The swaps page can be opened from three places:

1. The bottom sheet of main view:
<img width="330" alt="Screenshot 2024-06-20 at 11 27 52 AM" src="https://github.com/MetaMask/metamask-mobile/assets/3500406/0d8fee87-bbf2-4d3e-b449-375801cc884d">

2. The token page (for native or erc20):

<img width="332" alt="Screenshot 2024-06-20 at 11 29 59 AM" src="https://github.com/MetaMask/metamask-mobile/assets/3500406/b33f3f0d-24ac-43b9-bbdb-931e67824e4c">

3. The send flow, if you have insufficient balance:

<img width="303" alt="Screenshot 2024-06-20 at 11 30 50 AM" src="https://github.com/MetaMask/metamask-mobile/assets/3500406/f37881fe-d69b-48ba-bca8-c36668882e97">

It was partially differentiated before, with a `source` parameter set to `MainView` or `TokenView`.  But it was not differentiated for native tokens (always `MainView` even when from the token page).  And the send flow was never differentiated (always `MainView`).

Now the `source` parameters reflects these 3 pages with `MainView`, `TokenView`, or `SendFlow`.

## **Related issues**


## **Manual testing steps**

- Have metametrics enabled
- On the NFT tab, click an individual NFT
  - Verify in console the `Collectible Details Opened` event.
- On the NFT tab, long press an NFT > hide it
  - Verify in console the `Collectible Removed` event 
- Click action button at bottom of main view > swap button
  - Verify in console the `Swaps Opened` event with `source=MainView`
- On the tokens tab, click the native token > swap button
  - Verify in console the `Swaps Opened` event with `source=TokenView`
- On the tokens tab, click an ERC20 token > swap button
  - Verify in console the `Swaps Opened` event with `source=TokenView`
- Start to send an ERC20 token, but enter an amount more than you own
  - Click `Swap tokens` under the `insufficient funds` banner
  - Verify in console the `Swaps Opened` event with `source=SendFlow`

## **Screenshots/Recordings**

No visual differences, just different metrics emitted

### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
